internal/controller/nonadminbackup_controller.go

```diff
@@ -198,11 +480,17 @@ func (r *NonAdminBackupReconciler) validateSpec(ctx context.Context, logger logr
 //
 // This function generates a UUID and stores it in the VeleroBackup status field of NonAdminBackup.
 func (r *NonAdminBackupReconciler) setBackupUUIDInStatus(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	// If the NonAdminBackup is marked for deletion (either by DeleteBackup in the spec or with a DeletionTimestamp), skip setBackupUUIDInStatus
+	if nab.Spec.DeleteBackup || !nab.ObjectMeta.DeletionTimestamp.IsZero() {
+		return false, nil
+	}
+
 	if nab.Status.VeleroBackup == nil || nab.Status.VeleroBackup.NameUUID == constant.EmptyString {
 		veleroBackupNameUUID := function.GenerateNacObjectNameWithUUID(nab.Namespace, nab.Name)
 		nab.Status.VeleroBackup = &nacv1alpha1.VeleroBackup{
 			NameUUID:  veleroBackupNameUUID,
 			Namespace: r.OADPNamespace,
+			Name:      veleroBackupNameUUID,
```

Comment:
why duplicate the information?

Reply:
It is to keep consistency between Status fields for `VeleroBackup` and `VeleroDeleteBackupRequest`. We do not guarantee that Name is the key that links NAB with Velero. The field `NameUUID` is to be used as reference only, for example:

https://github.com/migtools/oadp-non-admin/pull/108/files/289367e71f9939098a18213c5205d6fa5a50cee6#diff-1fc992498fb761a2b0d175e637f5f945c2eb932ffacb0713d8724f4c14054c31R87

In the `VeleroBackup` we use the NameUUID, however in the `VeleroDeleteBackupRequest` we use name generated similarly how Velero would gneerate `DeleteBackupRequest` name, so we need to have both fields there. In case of `VeleroBackup` those fields will be the same, however as I wrote for consistency it's good in my opinion to have similar structure of the fields.

Reply:
yeah i think its time to rename the UUID field as I mentioned in the other comment.

Reply:
Done please check newest commit.